plugins/Jobs.cpp

```diff
@@ -403,8 +403,26 @@ bool BedrockPlugin_Jobs::processCommand(SQLite& db, BedrockCommand& command) {
         for (auto& job : jsonJobs) {
             // If unique flag was passed and the job exist in the DB, then we can finish the command without escalating to
             // master.
+
+            // If this is a mock request, we insert that into the data.
+            string originalData = job["data"];
+            if (command.request.isSet("mockRequest")) {
+                if (job["data"].empty()) {
+                    job["data"] = "{\"mockRequest\":\"true\"}";
+                } else {
+                    STable data = SParseJSONObject(job["data"]);
+                    data["mockRequest"] = "true";
+                    job["data"] = SComposeJSONObject(data);
+                }
+            }
+
             uint64_t updateJobID = 0;
             if (SContains(job, "unique") && job["unique"] == "true") {
+                // We don't try to handle unique jobs in mock requests.
```

Comment:
Why not?

Reply:
Because it's confusing without doing a lot of extra work. To elaborate on what @mea36 said - what happens when a "unique" job is first created by a mock request, and then a real request tries to create it afterward? It's possible to come up with a way to handle this, but it's not straightforward and it's not really required for the first iteration of this tester.